core/frontend/src/IModelConnection.ts

```diff
@@ -486,18 +528,57 @@ export abstract class IModelConnection extends IModel {
     return result;
   }
 
-  /** Convert a [[Cartographic]] to a point in this iModel's Spatial coordinates using the Geographic location services for this IModelConnection or [[IModel.ecefLocation]].
+  /** Convert a [Cartographic]($common) to a point in this iModel's Spatial coordinates using a [[GeoConverter]] or[[IModel.ecefLocation]($common).
    * @param cartographic A cartographic location
    * @param result If defined, use this for output
    * @returns A point in this iModel's spatial coordinates
    * @throws IModelError if [[isGeoLocated]] is false or cartographic location could not be converted.
-   * @see [[cartographicToSpatialFromGcs]]
-   * @see [[cartographicToSpatialFromEcef]]
+   * @see [[spatialFromCartographic]] to convert multiple points at once.
+   * @see [[cartographicToSpatialFromEcef]] to synchronously convert points using the iModel's ECEF transform.
    */
   public async cartographicToSpatial(cartographic: Cartographic, result?: Point3d): Promise<Point3d> {
     return (this.noGcsDefined ? this.cartographicToSpatialFromEcef(cartographic, result) : this.cartographicToSpatialFromGcs(cartographic, result));
   }
 
+  /** Convert [Cartographic]($common) coordinates into points in this iModel's spatial coordinate system using a [[GeoConverter]] or the iModel's [EcefLocation]($common).
+   * @param cartographic Coordinates to be converted to the iModel's spatial coordinate system.
+   * @returns The `cartographic` coordinates converted to spatial coordinates, of the same length and order as `cartographic`.
+   * @throws IModelError if [[isGeoLocated]] is false or any point could not be converted.
+   * @see [[cartographicFromSpatial]] to perform the inverse conversion.
+   */
+  public async spatialFromCartographic(cartographic: Cartographic[]): Promise<Point3d[]> {
```

Comment:
@pmconne 
If I got an array of array of points, do I need to serialize all points in a single flat array and then make a single call to `spatialFromCartographic`.

Also what would be a good way to verify how many calls to backend service was made if Im using display-test-app ?

Reply:
I already answered this.
There are tests included in PR that you can use as a guide if you're writing new ones.